corehq/apps/app_execution/data_model.py

```diff
@@ -98,11 +104,51 @@ def __str__(self):
         return f"Entity Select: {self.value}"
 
 
+@define
+class QueryInputValidationStep(Step):
+    type: ClassVar[str] = "query_input_validation"
+    is_form_step: ClassVar[bool] = False
+
+    inputs: dict
+    """Search inputs dict. Keys are field names and values are search values."""
+
+    def get_request_data(self, session, data):
+        query_key = session.data["queryKey"]
+        return {
+            **data,
+            # DataDog tag value
+            "requestInitiatedByTag": "field_change",
+            "query_data": {
+                query_key: {
+                    "execute": False,
+                    "force_manual_search": True,
+                    "inputs": self.inputs,
+                }
+            },
+        }
+
+
 @define
 class QueryStep(Step):
     type: ClassVar[str] = "query"
     is_form_step: ClassVar[bool] = False
+
     inputs: dict
+    """Search inputs dict. Keys are field names and values are search values."""
+
+    validate_inputs: bool = False
+    """Simulate a use updating search inputs on the UI. One request per update."""
```

Comment:
nit:
```suggestion
    """Simulate updating search inputs on the UI. One request per update."""
```
(or something like that - this line read weird to me)